### PR TITLE
League stats

### PR DESCRIPTION
**Pull request by jpc20**
Finish methods for league stats.
Will need to be refactored especially average_goals_by team method.
Also please take a look at the stubs to be sure those are implemented correctly.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -90,4 +90,65 @@ def average_goals_by_season
...
     end
   end
+
+  def count_of_teams
+    @teams.length
+  end
+
+  def average_goals_by_team(team_id, hoa = nil)
+    goals = 0
+    games = 0
```
Comment (jpc20) [inline lib/stat_tracker.rb:100]: Refactor

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -90,4 +90,65 @@ def average_goals_by_season
...
+    end
+    return 0 if games == 0
+    (goals.to_f / games.to_f).round(2)
+  end
+
+  def unique_team_ids
+    @game_teams.uniq { |game_team| game_team.team_id }.map{ |game_team| game_team.team_id }
+  end
+
+  def team_by_id(team_id)
```
Comment (Ashkanthegreat) [inline lib/stat_tracker.rb:120]: Maybe move this to a parent class?

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -90,4 +90,65 @@ def average_goals_by_season
...
+          goals += game_team.goals
+          games += 1
+      end
+    end
+    return 0 if games == 0
+    (goals.to_f / games.to_f).round(2)
+  end
+
+  def unique_team_ids
+    @game_teams.uniq { |game_team| game_team.team_id }.map{ |game_team| game_team.team_id }
```
Comment (arpariseau) [inline lib/stat_tracker.rb:117]: Either put uniq at the end of the map, or maybe do a group_by team_id and then call the key values?

File: test/stat_tracker_test.rb [hunk 4]
```diff
@@ -68,4 +68,57 @@ def test_it_return_season_average_goals
...
+
+  def test_it_can_find_unique_team_id
+    assert_equal [30, 52, 19, 23, 24, 4, 29, 12, 6, 17, 1, 2], @stat_tracker.unique_team_ids
+  end
+
+  def test_find_team_by_id
+    assert_equal 'Reign FC', @stat_tracker.team_by_id(54).team_name
+  end
+
+  def test_best_offense
```
Comment (Ashkanthegreat) [inline test/stat_tracker_test.rb:88]: Good use of stubs.

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -90,4 +90,65 @@ def average_goals_by_season
       average_of_goals_in_a_season(season.first.season)
     end
   end
+
+  def count_of_teams
+    @teams.length
+  end
+
+  def average_goals_by_team(team_id, hoa = nil)
```
Comment (Ashkanthegreat) [inline lib/stat_tracker.rb:98]: Maybe also move to a module.

**Pull request merged**